### PR TITLE
softmax: refine doc; input->x

### PR DESCRIPTION
**Pull request by zhupengyang**
### PR types

Others

### PR changes

APIs

### Describe

- update docs
- input -> x

image
image
image

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -1306,8 +1302,8 @@ def softmax(input, use_cudnn=False, name=None, axis=-1):
     attrs = {"axis": axis, "use_cudnn": use_cudnn}
 
     helper = LayerHelper('softmax', **locals())
-    check_variable_and_dtype(input, 'input', ['float16', 'float32', 'float64'],
-                             'softmax')
+    check_variable_and_dtype(input, 'input/x',
```
Comment (wangchaochaohu) [inline python/paddle/fluid/layers/nn.py:1306]: 这里可以通过栈信息看出来吧 不用强调x

File: python/paddle/nn/functional/activation.py [hunk 2]
```diff
@@ -305,6 +305,120 @@ def sigmoid(input, inplace=False, name=None):
...
+
+    2. Then ``x`` will be logically flattened to a 2-D matrix. The matrix's second
+    dimension(row length) is the same as the dimension :attr:`axis` of ``x``,
+    and the first dimension(column length) is the product of all other dimensions
+    of ``x``. For each row of the matrix, the softmax operator squashes the
+    K-dimensional(K is the width of the matrix, which is also the size of ``x``'s
+    dimension :attr:`axis`) vector of arbitrary real values to a K-dimensional
+    vector of real values in the range [0, 1] that add up to 1.
+
+    3. After the softmax operation is completed, the inverse operations of steps 1 and 2
```
Comment (wangchaochaohu) [inline python/paddle/nn/functional/activation.py:322]: and step 2

File: python/paddle/nn/functional/activation.py [hunk 3]
```diff
@@ -305,6 +305,120 @@ def sigmoid(input, inplace=False, name=None):
...
+    are performed to restore the two-dimensional matrix to the same dimension as the x.
```
Comment (wangchaochaohu) [inline python/paddle/nn/functional/activation.py]: ``x``

File: python/paddle/nn/functional/activation.py (hunk 2 above)
Comment (zhupengyang) [reply python/paddle/nn/functional/activation.py:322]: 这样也是可以的

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/nn.py:1306]: fluid.layers.softmax和F.softmax的输入名称不一样了，这样更好一些

File: python/paddle/nn/functional/activation.py (hunk 3 above)
Comment (zhupengyang) [reply python/paddle/nn/functional/activation.py]: Done

**Review by hong19860320 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -1190,10 +1190,6 @@ def chunk_eval(input,
 
 def softmax(input, use_cudnn=False, name=None, axis=-1):
     """
-    :alias_main: paddle.nn.functional.softmax
-	:alias: paddle.nn.functional.softmax,paddle.nn.functional.activation.softmax
-	:old_api: paddle.fluid.layers.softmax
-
     This operator implements the softmax layer. The calculation process is as follows:
 
     1. The dimension :attr:`axis` of the ``input`` will be permuted to the last.
```
Comment (hong19860320) [inline python/paddle/fluid/layers/nn.py:1195]: 这个地方的"input"是不是也得改下

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -1190,10 +1190,6 @@ def chunk_eval(input,
 
 def softmax(input, use_cudnn=False, name=None, axis=-1):
```
Comment (hong19860320) [inline python/paddle/fluid/layers/nn.py:1191]: input不用改成x吗？

File: python/paddle/nn/functional/activation.py [hunk 6]
```diff
@@ -305,6 +305,121 @@ def sigmoid(input, inplace=False, name=None):
...
+        x = paddle.imperative.to_variable(x)
+        out = F.softmax(x)
+        # [[[0.0320586 , 0.08714432, 0.23688282, 0.64391426],
+        #   [0.0320586 , 0.08714432, 0.23688282, 0.64391426],
+        #   [0.07232949, 0.19661193, 0.19661193, 0.53444665]],
+        # [[0.0320586 , 0.08714432, 0.23688282, 0.64391426],
+        #   [0.0320586 , 0.08714432, 0.23688282, 0.64391426],
+        #   [0.0320586 , 0.08714432, 0.23688282, 0.64391426]]]
+    """
+    return paddle.fluid.layers.softmax(input=x, axis=axis, name=name)
```
Comment (hong19860320) [inline python/paddle/nn/functional/activation.py:421]: 这个地方的input需要改吗？

File: python/paddle/fluid/layers/nn.py (hunk 4 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/nn.py:1195]: fluid下的不改，新的接口已经改了

File: python/paddle/fluid/layers/nn.py (hunk 5 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/nn.py:1191]: 同上

File: python/paddle/nn/functional/activation.py (hunk 6 above)
Comment (zhupengyang) [reply python/paddle/nn/functional/activation.py:421]: 同上

File: python/paddle/fluid/tests/unittests/test_softmax_op.py [hunk 7]
```diff
@@ -220,5 +221,46 @@ def get_x_shape(self):
         return [2, 3, 4, 5]
 
 
+class TestNnFunctionalSoftmaxApi(unittest.TestCase):
+    def setUp(self):
+        self.place = paddle.CUDAPlace(0) if core.is_compiled_with_cuda(
+        ) else paddle.CPUPlace()
+        self.x_np = np.array([[[2.0, 3.0, 4.0, 5.0], [3.0, 4.0, 5.0, 6.0],
```
Comment (T8T9) [inline python/paddle/fluid/tests/unittests/test_softmax_op.py:228]: 之前说输入数据使用固定种子，随机生成，总维度要在100以上。https://github.com/PaddlePaddle/Paddle/wiki/OP-test-input-shape-requirements

File: python/paddle/fluid/layers/nn.py [hunk 8]
```diff
@@ -1307,8 +1303,8 @@ def softmax(input, use_cudnn=False, name=None, axis=-1):
     attrs = {"axis": axis, "use_cudnn": use_cudnn}
 
     helper = LayerHelper('softmax', **locals())
-    check_variable_and_dtype(input, 'input', ['float16', 'float32', 'float64'],
```
[the remark this answers is not in the document]
Comment (zhupengyang) [reply python/paddle/fluid/layers/nn.py:1310]: 应该是和API显示的输入名称一致吧。因为现在是F.softmax调用的原来的softmax（原来是input，现在是x），所以报错信息的input_name调整为 "input/x" 。

File: python/paddle/nn/functional/activation.py [hunk 9]
```diff
@@ -305,6 +305,121 @@ def sigmoid(input, inplace=False, name=None):
...
+                         [0.01786798, 0.01786798, 0.04661262, 0.04661262],
+                         [0.97555875, 0.97555875, 0.93623955, 0.93623955]],
+                        [[0.00490169, 0.00490169, 0.00490169, 0.00490169],
+                         [0.26762315, 0.26762315, 0.26762315, 0.26762315],
+                         [0.72747516, 0.72747516, 0.72747516, 0.72747516]]]
+
+    Args:
+        x (Tensor): The input multi-dimension Tensor with data type float32, float64.
+        axis (int, optional): The axis along which to perform softmax calculations.
+            It should be in range [-R, R), where R is Rank(x). When ``axis`` < 0,
```
[the remark this answers is not in the document]
Comment (zhupengyang) [reply python/paddle/nn/functional/activation.py]: Done

File: python/paddle/fluid/tests/unittests/test_softmax_op.py (hunk 7 above)
Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/test_softmax_op.py]: Done

**Comment by T8T9** [comment]
LGTM

**Review by hong19860320 — APPROVED** [review]
LGTM

**Review by swtkiwi — APPROVED** [review]
LGTM
